utils/clients/queue_client.py

```diff
@@ -57,7 +58,12 @@ def disconnect(self):
         """
         logging.info("Disconnecting from activemq")
         if self._connection is not None and self._connection.is_connected():
-            self._connection.disconnect()
+            # By passing a receipt Stomp will call stop on the transport layer
+            # which causes it to wait on the listener thread (if it's still
+            # running). Without this step we just drop out, so the behaviour
+            # is not guaranteed. UUID is used by Stomp if we don't pass in
+            # a receipt so this matches the behaviour under the hood
```

Comment:
NITPICK: Because of the size of this comment, could we not just turn it into a docstring?

Reply:
In this case, no, since it's an implementation detail of stomp and shouldn't really be documented in the API doc string IMO